### PR TITLE
Enhance/#8795 - Implement RRM `findMatchedPublication()` action

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #8795 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/reader-revenue-manager/datastore/constants.js [hunk 1]
```diff
@@ -19,3 +19,9 @@
 export const MODULES_READER_REVENUE_MANAGER = 'modules/reader-revenue-manager';
 
 export const ERROR_CODE_NON_HTTPS_SITE = 'non_https_site';
+
+export const PUBLICATION_ONBOARDING_STATE = {
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/datastore/constants.js:23]: What do you think about changing this to `PUBLICATION_ONBOARDING_STATES` (plural)?

**Review by nfmohit — CHANGES_REQUESTED** [review]
Excellent work on this, @hussain-t ! I have left a very small suggestion, after addressing which, we should be able to merge this. Please let me know what you think, thanks!

File: assets/js/modules/reader-revenue-manager/datastore/constants.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/modules/reader-revenue-manager/datastore/constants.js:23]: Given that only one state can be referenced at a time for a publication, I thought the singular form was appropriate. I can change it if you prefer the plural.

Comment (hussain-t) [reply assets/js/modules/reader-revenue-manager/datastore/constants.js]: Updated ✅

**Review by nfmohit — APPROVED** [review]
LGTM, thanks @hussain-t !

**Pull request merged**